org-code-javabuilder/README.md

```diff
@@ -1,33 +1,138 @@
-# Javabuilder App
-### Local Development
-To run Javabuilder in isolation, run the `LocalMain` class from an IDE (IntelliJ is recommended).
-This will exercise the compile and run logic in isolation. Javabuilder will execute the
-program in src/main/resources/main.json. You can update which file is used from `LocalProjectFileManager`.
-Input and output will be directed to the terminal.
-Any non-Java file used or created by the program (for example `grid.txt` in `main_painter.json`) will be
-created in the `org-code-javabuilder` folder when running locally, so be aware you may want 
-to do some cleanup after running code that relies on text files (in our Lambda runtime the code 
-all runs from a temporary folder and handles this cleanup). 
-
-### Developing with Dashboard
+# Javabuilder build-and-run package
+This portion of Javabuilder contains the code that compiles and runs student code as well
+as the code for the `org.code` Java packages that are used in the Java Lab mini apps.
+
+### Directory
+* [lib](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/lib)
+  contains the "backend" code that compiles, builds, and runs student code. This is not
+  user-facing. It also handles the WebSocket layer that communicates between the client
+  and the running program. Also contained here are implementations of the communication
+  protocol and program initialization that are specific to local development and AWS
+  development. Local implementations are in the `dev.javabuilder` package while AWS
+  implementations are in the main `org.code.javabuilder` package and are prefixed with
+  `AWS` or `Lambda`
+* [media](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/media)
+  contains a package that can be imported by the user to edit sounds, colors, fonts, and
+  images.
+* [neighborhood](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/neighborhood)
+  contains a package that can be imported by the user to interact with the neighborhood
+  mini-app, a maze-like app created for AP CSA.
+* [protocol](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/protocol)
+  contains the communication protocol that handles messages between Java Lab (client) and
+  Javabuilder (server)
+* [theater](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/theater)
+  contains a package that can be imported by the user to interact with the theater 
+  mini-app, an image and sound editing app created for AP CSA and used in standalone Java
+  Lab projects.
+
+## Local Development
+### Setup your project
+1. Install a Java IDE - We recommend IntelliJ Community edition.
+    1. If given the option when installing, select Java 11 as the default
+1. If Java 11 was not installed as part of the IDE, install 
+   [OpenJDK 11](https://jdk.java.net/java-se-ri/11)
+1. Download the repo and build. Building is a one-time step that seeds the git commit
```

Comment:
if using IntelliJ, i think it's also worth calling out that the project should be imported as a gradle project so it gets setup correctly in the IDE (File -> New -> Project From Existing Sources I believe)

Reply:
👍